airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -186,18 +186,18 @@ def ti_run(
         if not dr:
             raise ValueError(f"DagRun with dag_id={ti.dag_id} and run_id={ti.run_id} not found.")
 
-        # Clear XCom data for the task instance since we are certain it is executing
+        # Send the XCom data for clearance for the task instance since we are certain it is executing at this point.
         # However, do not clear it for deferral
         if not ti.next_method:
             map_index = None if ti.map_index < 0 else ti.map_index
             log.info("Clearing xcom data for task id: %s", ti_id_str)
```

Comment:
This log isn't accurate anymore - maybe not worth keeping since we aren't doing anything destructive/of note here.

Reply:
Agreed, removed it